rascal-vscode-extension/src/auto-jvm/downloaders.ts

```diff
@@ -181,7 +182,33 @@
     return `jdk-${rel.openjdk_version}`;
 }
 
-function mapTemuringCorrettoArch(): TemurinArchitectures {
+export async function identifyLatestCorrettoRelease(version: number, arch: TemurinArchitectures, platform: CorrettoPlatforms): Promise<string> {
+    const url = `https://corretto.aws/downloads/latest/amazon-corretto-${version}-${arch}-${platform}-jdk.${platform === "windows" ? "zip" : "tar.gz"}`;
+    const response = await fetch(url, {method: 'HEAD'});
+    const fileUrl = response.url;
+    const versionPattern = /resources\/((\d*\.?)+)\//;
```

Comment:
I think the regular expression has a bit too many optional parts. since `\.?` can match 0 or 1 `.` and `\d*` can match zero or more chars. maybe we can write it more like: `/resources\/[0-9\.]+\//`?